lib/beaker/hypervisor/docker.rb

```diff
@@ -65,6 +65,19 @@ def provision
         host['docker_container'] = container
         host['docker_image'] = image
       end
+
+      hack_docker_etc_hosts @hosts, @options
+
+    end
+
+    def hack_docker_etc_hosts hosts, opts
+      etc_hosts = "127.0.0.1\tlocalhost localhost.localdomain\n"
+      hosts.each do |host|
+        etc_hosts += "#{host['docker_container'].json["NetworkSettings"]["IPAddress"]}\t#{host.name}\n"
```

Comment:
Looking at https://github.com/madAndroid/beaker/blob/hack_etc_hosts_docker/lib/beaker/hypervisor/docker.rb#L56 it looks like host['ip'] is being set per-host already, so you should be able to use the pre-existing hack_etc_host method.